src/components/BrowserFilter/BrowserFilter.react.js

```diff
@@ -46,13 +46,18 @@ export default class BrowserFilter extends React.Component {
   toggle() {
     let filters = this.props.filters;
     if (this.props.filters.size === 0) {
-      const available = Filters.availableFilters(
-        this.props.schema,
-        null,
-        this.state.blacklistedFilters
+      const available = Filters.findRelatedClasses(
+        this.props.className,
+        this.props.allClassesSchema,
+        this.state.blacklistedFilters,
+        this.state.filters
       );
-      const field = Object.keys(available)[0];
-      filters = new List([new Map({ field: field, constraint: available[field][0] })]);
+      const filterClass = Object.keys(available)[0];
+      const filterField = Object.keys(available[filterClass])[0];
+      const filterConstraint = available[filterClass][filterField][0];
+      filters = new List([
+        new Map({ class: filterClass, field: filterField, constraint: filterConstraint }),
+      ]);
```

Comment:
This code block seems to occur almost identical in `addRow`. Could you refactor these occurrences to deduplicate? Maybe keep the `availableFilters` method and move the code block there?

Reply:
> Maybe keep the availableFilters method

maybe i can make a function like this 
```
export function getFilterDetails(available) {
  const filterClass = Object.keys(available)[0];
  const filterField = Object.keys(available[filterClass])[0];
  const filterConstraint = available[filterClass][filterField][0];
  return { filterClass, filterField, filterConstraint };
}
```

and call it here like

```
addRow() {
    const available = Filters.findRelatedClasses(
      this.props.className,
      this.props.allClassesSchema,
      this.state.blacklistedFilters,
      this.state.filters
    );
    const { filterClass, filterField, filterConstraint } = Filters.getFilterDetails(available);
    this.setState(({ filters }) => ({
      filters: filters.push(
        new Map({ class: filterClass, field: filterField, constraint: filterConstraint })
      ),
      editMode: true,
    }));
  }
  ```
  if you think this works , i can push the changes

Reply:
I haven't looked at the logic; I just notice that there is duplicate code which maybe could be optimized. If your suggestion works and it removes the duplication, please go ahead.

Reply:
actually the duplicate code is nothing just getting the class , field and constraint value from the available. 

I made a function which takes the available and returns a object which contain of class , field and this makes the code more neat in my take and removes the duplication